packages/curl.rb

```diff
@@ -3,108 +3,237 @@
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm", "#{@deppath}/include/"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm-generic", "#{@deppath}/include/"
+
+    @arch_ssp_cflags = ''
+    case ARCH
+    when 'i686'
+      @arch_ssp_cflags = '-fno-stack-protector'
+    end
+
+    @curldep_cmake_options = "CC=musl-gcc \
+          CFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CPPFLAGS='-I#{@deppath}/include' \
+          LDFLAGS='-L#{@deppath}/lib' \
+          cmake \
+          -DCMAKE_INSTALL_PREFIX='#{@deppath}' \
+          -DCMAKE_LIBRARY_PATH='#{@deppath}/lib' \
+          -DCMAKE_C_COMPILER=`which musl-gcc` \
+          -DCMAKE_INCLUDE_DIRECTORIES_BEFORE=ON \
+          -DINCLUDE_DIRECTORIES=#{@deppath}/include \
+          -DCMAKE_C_FLAGS='-O3 -fPIC -pipe -flto -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          -DCMAKE_EXE_LINKER_FLAGS='-flto' \
+          -DCMAKE_SHARED_LINKER_FLAGS='-flto' \
+          -DCMAKE_STATIC_LINKER_FLAGS='-flto' \
+          -DCMAKE_MODULE_LINKER_FLAGS='-flto' \
+          -DPROPERTY_INTERPROCEDURAL_OPTIMIZATION=TRUE \
+          -DCMAKE_BUILD_TYPE=Release"
+
+    @curldep_env_options = "CC=musl-gcc \
+        CFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+        CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans#{@arch_ssp_cflags}' \
+        CPPFLAGS='-I#{@deppath}/include' \
+        LDFLAGS='-L#{@deppath}/lib'"
+
+    FileUtils.mkdir_p 'build/zlib'
+    Dir.chdir 'build/zlib' do
+      puts 'Building Zlib.'.lightgreen
+      system 'curl -Ls http://zlib.net/zlib-1.2.11.tar.gz | tar --strip-components=1 -zxf -'
+      system "#{@curldep_env_options} ./configure --prefix=#{@deppath} \
+        --static"
+      system 'make'
+      system 'make install'
+    end
+    FileUtils.mkdir_p 'build/zstd'
+    Dir.chdir 'build/zstd' do
+      puts 'Building Zstd.'.lightgreen
+      system 'curl -Ls https://github.com/facebook/zstd/archive/v1.5.0.tar.gz | tar --strip-components=1 -zxf -'
+      FileUtils.mkdir('build/cmake/builddir')
+      Dir.chdir('build/cmake/builddir') do
+        system "#{@curldep_cmake_options} \
+        -DZSTD_BUILD_STATIC=ON \
+        -DZSTD_BUILD_SHARED=OFF \
+        ../ -G Ninja"
+      end
+      system 'ninja -C build/cmake/builddir'
+      system 'ninja -C build/cmake/builddir install'
+    end
+    FileUtils.mkdir_p 'build/brotli'
+    Dir.chdir 'build/brotli' do
+      puts 'Building Brotli.'.lightgreen
+      system 'curl -Ls https://github.com/google/brotli/archive/v1.0.9.tar.gz | tar --strip-components=1 -zxf -'
```

Comment:
Why not use the `brotli` package instead?

Reply:
These are built with musl. Static libraries built with gcc or clang appear to sometimes still invoke glibc 2.2x, and that's creating subtle libnss crashes during dns lookups with curl. Compiling these separately with musl avoids that problem entirely.

Also, curl no longer segfaults.

See https://github.com/curl/curl/issues/7194 for the issue I opened before I gave up on compiling with the chromebrew static libraries.

Reply:
Why don't we build the crew packages with musl?  Seems like a maintenance nightmare to do it this way.

Reply:
Musl is ONLY being used for the static curl dependencies, because it's not breaking openssl in the way that glibc 2.27 appears to be doing.

Even libcurl (and static git, which uses it) are being built against glibc.

This is probably a decent explainer on why not to build everything against musl: https://www.madetech.com/blog/a-tale-in-adopting-alpine-linux-for-docker-problems-we-faced-with-rspec-testing/

(Also, if we decide to build everything against a newer libc, we should probably just use a newer glibc...)